homeassistant/components/lupusec.py

```diff
@@ -0,0 +1,91 @@
+"""
+This component provides basic support for Lupusec Home Security system.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/lupusec
+"""
+
+from datetime import timedelta
+import logging
+import voluptuous as vol
+
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers import discovery
+from homeassistant.const import (CONF_USERNAME, CONF_PASSWORD,
+                                 CONF_NAME, CONF_IP_ADDRESS,
+                                 CONF_SCAN_INTERVAL)
+from homeassistant.helpers.entity import Entity
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['lupupy==0.0.5']
+
+DOMAIN = 'lupusec'
+DEFAULT_SCAN_INTERVAL = timedelta(seconds=2)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_IP_ADDRESS): cv.string,
+        vol.Optional(CONF_NAME): cv.string,
+        vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL):
+            cv.time_period,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+SCAN_INTERVAL = timedelta(seconds=2)
+
+LUPUSEC_PLATFORMS = [
+    'alarm_control_panel', 'binary_sensor', 'switch'
+]
+
+
+
+
+def setup(hass, config):
+    """Set up Lupusec component."""
+    from lupupy.exceptions import LupusecException
+
+    conf = config[DOMAIN]
+    username = conf[CONF_USERNAME]
+    password = conf[CONF_PASSWORD]
+    ip_address = conf[CONF_IP_ADDRESS]
+    name = conf.get(CONF_NAME)
+    SCAN_INTERVAL = conf.get(CONF_SCAN_INTERVAL)
+
+    try:
+        hass.data[DOMAIN] = LupusecSystem(username, password, ip_address, name)
+    except LupusecException as ex:
+        _LOGGER.warning(ex)
```

Comment:
We should validate that credentials are correct, otherwise log error and `return False` from component setup.

Reply:
Added a notification and false returning.

Reply:
Log error instead of warning.